src/map/npc.cpp

```diff
@@ -352,6 +354,26 @@ uint64 StylistDatabase::parseBodyNode( const ryml::NodeRef& node ){
 			entry->value = value;
 		}
 
+#if PACKETVER >= 20231220
+		if (constant == LOOK_BODY2 && this->nodeExists(node, "RequiredJob")) {
+			const ryml::NodeRef& jobNode = optionNode["RequiredJob"];
+
+			for (const auto& jobit : jobNode) {
+				std::string jobName;
+				c4::from_chars(jobit.key(), &jobName);
+				std::string job_name_constant = "JOB_" + jobName;
+				int64 job_id;
+
+				if (!script_get_constant(job_name_constant.c_str(), &job_id)) {
+					this->invalidWarning(optionNode["RequiredJob"], "Job %s does not exist.\n", jobName.c_str());
+					return 0;
+				}
+
+				entry->required_job.push_back(job_id);
```

Comment:
Missing true/false logic to add (if not already in) or remove the job_id from the list